app/models/region.rb

```diff
@@ -0,0 +1,51 @@
+class Region < ApplicationRecord
+  ltree :path
+  extend FriendlyId
+  friendly_id :name, use: :slugged
+
+  validates :name, presence: true
+  validates :slug, presence: true, uniqueness: true
+  validates :path, presence: true
+
+  belongs_to :kind, class_name: "RegionKind", foreign_key: "region_kind_id"
+  belongs_to :source, polymorphic: true, optional: true
+
+  before_discard do
+    self.path = nil
+  end
+
+  def self.create_region_from(parent:, kind:, name: nil, source: nil)
+    raise ArgumentError, "Provide either a name or a source" if (name && source) || (name.blank? && source.blank?)
+    region_name = name || source.name
+    region = Region.new name: region_name, kind: kind
+    region.send :set_slug
+    region.source = source if source
+    region.path = "#{parent.path}.#{region.slug.tr("-", "_")}"
+    region.save!
+    region
+  end
+
+  def self.backfill!
+    instance_kind = RegionKind.find_by!(name: "Instance")
+    org_kind = instance_kind.children.first
+    facility_group_kind = org_kind.children.first
+    block_kind = facility_group_kind.children.first
+    facility_kind = block_kind.children.first
+
+    instance = Region.create! name: "India", path: "India", kind: instance_kind
+    Organization.all.each do |org|
+      org_region = create_region_from(source: org, kind: org_kind, parent: instance)
+
+      org.facility_groups.each do |facility_group|
+        facility_group_region = create_region_from(source: facility_group, kind: facility_group_kind, parent: org_region)
+
+        facility_group.facilities.group_by(&:zone).each do |block_name, facilities|
+          block_region = create_region_from(name: block_name, kind: block_kind, parent: facility_group_region)
```

Comment:
We probably don't want to create blocks until the data is cleaned up on prod, but I wanted to include it here to see how it would work.